bluetooth_mesh/messages/capnproto.py

```diff
@@ -0,0 +1,314 @@
+    GreedyBytes,
+    GreedyRange,
+    NullStripped,
+    Padded,
+    Pass,
+    Rebuild,
+    Renamed,
+    Restreamed,
+    Select,
+    StopIf,
+    Struct,
+    Switch,
+    Transformed,
+)
+
+from bluetooth_mesh.messages import AccessMessage
+
+
+class VariableNames:
+    def __init__(self, root_name):
+        self.names = {}
+        self._load_names(root_name)
+
+    def _load_names(self, root_name):
+        root_module = importlib.import_module(root_name)
+
+        for loader, name, is_pkg in pkgutil.walk_packages(root_module.__path__):
+            module_name = f"{root_name}.{name}"
+            module = importlib.import_module(module_name)
+            self.names.update({id(v): k for k, v in module.__dict__.items()})
+
+            if is_pkg:
+                self._load_names(module_name)
+
+    def __getitem__(self, item):
+        return self.names.get(id(item))
+
+
+names = VariableNames("bluetooth_mesh.messages")
+
+
+def convert(
+    con,
+    visitor,
+    field_name=None,
+    struct_name=None,
+    many=False,
+    message_name=None,
+):
+    struct_name = struct_name or names[con] or f"{message_name}Params"
+
+    if field_name and field_name.startswith("_"):
+        return
+
+    if isinstance(con, Struct):
+        visitor.enter_struct(field_name, struct_name, many=many)
+
+        for subcon in con.subcons:
+            convert(subcon, visitor, message_name=message_name)
+
+        visitor.exit()
+
+    elif isinstance(con, Switch):
+        if many:
+            visitor.enter_struct(field_name, struct_name, many=many)
+            visitor.enter_union(None)
+        else:
+            visitor.enter_union(field_name)
+
+        for key, subcon in con.cases.items():
+            if isinstance(key, IntEnum):
+                key = type(key)(key).name
+
+            convert(subcon, visitor, field_name=key, message_name=message_name)
+
+        visitor.exit()
+
+        if many:
+            visitor.exit()
+
+    elif isinstance(con, Select):
+        if many:
+            visitor.enter_struct(field_name, struct_name, many=many)
+            visitor.enter_union(None)
+        else:
+            visitor.enter_union(field_name)
+
+        for subcon in con.subcons:
+            convert(
+                subcon,
+                visitor,
+                message_name=message_name,
+            )
+
+        visitor.exit()
+
+        if many:
+            visitor.exit()
+
+    elif isinstance(con, Renamed):
+        convert(
+            con.subcon,
+            visitor,
+            field_name=con.name,
+            message_name=message_name,
+            many=many,
+        )
+
+    elif isinstance(con, (GreedyRange, Array)):
+        convert(
+            con.subcon,
+            visitor,
+            field_name=field_name,
+            # struct_name=struct_name,
+            many=True,
+            message_name=message_name,
+        )
+
+    elif con is Pass:
+        return
+
+    elif isinstance(con, StopIf):
```

Comment:
IMO if this subcon is present there should be two types of messages created (one contains fields up to this marker) and the second one contains all fields.